File: game/guess-number/guess-number.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main (int argc, char const* argv[])
{
    int ans, cur;

    printf("Game Start!\n");
    printf("Loading...\n");
    srand((unsigned)time(0));
    ans = rand() % 1000;
    printf("Ready! Go!\n");

    while (1) {
        printf("Guess Number: ");
        scanf("%d", &cur);

        if (cur > ans) {
            printf("Your number is bigger than the answer.\n");
        } else if (cur < ans) {
            printf("Your number is smaller than the answer.\n");
        } else {
            break;
        }
    }

    printf("You Got It!\n");
    printf("The answer is %d.\n", ans);

    return 0;
}

File: game/guess-number/guess-number.pas
program guess_number;
var
    ans, cur: longint;
begin
    assign(input, '');
    assign(output, '');
    reset(input);
    rewrite(output);

    writeln('Game Start!');
    writeln('Loading...');
    randomize;
    ans := random(1000) + 1;
    writeln('Ready! Go!');

    while (true) do
    begin
        write('Guess Number: ');
        readln(cur);

        if (cur > ans) then writeln('Your number is bigger than the answer.')
        else if (cur < ans) then writeln('Your number is smaller than the answer.')
        else break;
    end;

    writeln('Got It!');
    writeln('The answer is ', ans);

    close(input);
    close(output);
end.
